app/models/user.rb

```diff
@@ -101,6 +101,12 @@ def pending_profile
     profiles.gpo_verification_pending.order(created_at: :desc).first
   end
 
+  def personal_key_generated_at
+    encrypted_recovery_code_digest_generated_at ||
+      active_profile&.verified_at ||
```

Comment:
I have a light suspicion that we may not need to consider `active_profile` at all anymore with the new logic to use whichever verified profile was most recently activated, since it shouldn't be possible to have an older profile be considered the "active" profile. That being said, with all the different possible profile states, it might be safest to keep it this way.